### PR TITLE
Feature/messages

### PR DESCRIPTION
**Pull request by Pavliv**
Need some fixes, but basic work done.

### CODE REVIEW

File: app/assets/javascripts/messages.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+$(document).on('ready page:load', function (){
+	$("#edit-message").on('click', function(){
+		var text = $("span#message-text-field").text();
+		$("span#message-text-field").replaceWith('<div class="input-group">' +
+      	'<input id="new-message-field" type="text" class="form-control" value="'+text+'"><span class="input-group-btn">' +
+        '<button id="confirm-message-changes" class="btn btn-secondary" type="button">Ok</button></span></div>'
+      );
+		$("#confirm-message-changes").on('click', function(){
```
Comment (yasinishyn) [inline app/assets/javascripts/messages.js:8]: Message id is saved in span text?

File: app/assets/javascripts/messages.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	      },
+	      dataType: "json",
+	      type: "PUT",
+	      success: function(data){
+	        $("#new-message-field").closest('div.input-group').replaceWith('<span id="message-text-field">'+new_text+'</span>');
+	      }
+	    });
+		});
+	});
+
```
Comment (yasinishyn) [inline app/assets/javascripts/messages.js:25]: What are those white spaces? lines: 26-43

File: app/assets/stylesheets/messages.scss [hunk 3]
```diff
@@ -1,3 +1,124 @@
...
-// You can use Sass (SCSS) here: http://sass-lang.com/
+$dark_grey:#3f4a58;
+$orange:#f3795b;
+$green:#3c763d;
+$light_green:#d6e9c6;
+$red:#a94442;
+$light_red:#ebccd1;
+$yellow:#f0c036;
+$grey:#969696;
+
```
Comment (yasinishyn) [inline app/assets/stylesheets/messages.scss:9]: User 2 spaces for tabs.

File: app/assets/stylesheets/messages.scss [hunk 4]
```diff
@@ -1,3 +1,124 @@
...
+		padding: 5px;
+	}
+}
+
+.msg-frm{
+	margin: 35px;	
+}
+
+.messages-avatar-male, .messages-avatar-female{
+	display: inline-block;
```
Comment (yasinishyn) [inline app/assets/stylesheets/messages.scss:119]: Why do you need commented code `// float: left;` here?

File: app/controllers/messages_controller.rb [hunk 5]
```diff
@@ -67,6 +81,10 @@ def set_message
       @message = Message.find(params[:id])
     end
 
+    def set_users
```
Comment (yasinishyn) [inline app/controllers/messages_controller.rb]: `@users = User.where(role: User.roles[ current_user.rider? :driver : :rider])

File: app/controllers/orders_controller.rb [hunk 6]
```diff
@@ -32,6 +32,7 @@ def new
 
       @marker_options = l1.marker_params
       @map_options = l1.map_params
```
Comment (yasinishyn) [inline app/controllers/orders_controller.rb:61]: This code is duplicated in both `app/controllers/messages_controller.rb` and `app/controllers/orders_controller.rb`. Move logic, to User model.

`current_user.oposite_roles`

and inside the model

```
def oposite_roles
  User.where(role: User.roles[rider? :driver : :rider])
end
```

Therefore in controller

`
 @users = current_user.oposite_roles
`

File: app/helpers/messages_helper.rb [hunk 7]
```diff
@@ -1,2 +1,21 @@
 module MessagesHelper
```
Comment (yasinishyn) [inline app/helpers/messages_helper.rb:1]: Use 2 spaces for tabs.

File: app/models/message.rb [hunk 8]
```diff
@@ -11,4 +11,5 @@
 #
```
Comment (yasinishyn) [inline app/models/message.rb:11]: Use 2 spaces for tabs.

File: app/views/layouts/_chat_panel.html.slim [hunk 9]
```diff
@@ -3,15 +3,13 @@ button.toggle-messages.btn#pollSlider-button
...
-    .messages-avatar-female
-    h4 Sheyla
-    p Hi There...
-  button.messages-invite-friend.btn.btn-style Invite Friend
+  .related-block
+    - @users.each do |user|
+      .messages-box
+        .messages-avatar-male
+        h4 = user.name.capitalize
+        = link_to "conversation", "/messages/user/#{user.id}", class:('btn-style btn message-btn')
```
Comment (yasinishyn) [inline app/views/layouts/_chat_panel.html.slim:11]: Why do you need commented html here `/ button.messages-invite-friend.btn.btn-style Invite Friend` ?

File: app/views/messages/_form.html.slim [hunk 10]
```diff
@@ -1,10 +1,15 @@
 = simple_form_for(@message) do |f|
   = f.error_notification
-
   .form-inputs
-    = f.input :from_id
-    = f.input :to_id
-    = f.input :text
+  	.form-group
```
Comment (yasinishyn) [inline app/views/messages/_form.html.slim:4]: Why do you need commented html here?

File: app/views/messages/_form.html.slim [hunk 11]
```diff
@@ -1,10 +1,15 @@
...
+  		/ | From:
+  		/ = current_user.name.capitalize
+    = f.hidden_field :from_id, value: current_user.id
+   	.form-group
```
Comment (yasinishyn) [inline app/views/messages/_form.html.slim:6]: Why do you need commented html here?

File: app/views/messages/_message_box.html.slim [hunk 12]
```diff
@@ -0,0 +1,37 @@
+.row.message-box
```
Comment (yasinishyn) [inline app/views/messages/_message_box.html.slim:1]: Why do you need commented html here?

File: app/views/messages/_message_box.html.slim [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+	/ span#message-id
```
Comment (yasinishyn) [inline app/views/messages/_message_box.html.slim]: Use 2 spaces for tabs.

File: app/views/messages/index.html.slim [hunk 14]
```diff
@@ -1,25 +1,5 @@
...
-        td = message.from_id
-        td = message.to_id
-        td = message.text
-        td = link_to 'Show', message
-        td = link_to 'Edit', edit_message_path(message)
-        td = link_to 'Destroy', message, data: {:confirm => 'Are you sure?'}, :method => :delete
-
-br
-
-= link_to 'New Message', new_message_path
```
Comment (yasinishyn) [inline app/views/messages/index.html.slim:25]: Why do you need commented html here?

File: app/views/messages/show.html.slim [hunk 15]
```diff
@@ -1,15 +1,18 @@
...
-  strong To:
-  = @message.to_id
-p
-  strong Text:
-  = @message.text
-
-= link_to 'Edit', edit_message_path(@message)
-'|
-= link_to 'Back', messages_path
+/ p#notice = notice
```
Comment (yasinishyn) [inline app/views/messages/show.html.slim]: Why do you need commented html here?
 Add a line note

**Comment by arsenslyusarchuk** [comment]
Starting conversation, received an error: undefined method `to_id' for nil:NilClass in 'user_index.html.slim'

`= link_to find_user(@messages.first.to_id).name.capitalize , {controller: "profiles", action: "show", id: find_user(@messages.first.to_id).id}`

**Pull request merged**